api/api.go

```diff
@@ -37,9 +37,9 @@ func HttpPOST(url string, data string) (resp *http.Response, err error) {
 
 // handleHTTPRequest is a wrapper function that handles all incoming calls to
 // the API.
-func handleHTTPRequest(mux *http.ServeMux, url string, handler http.HandlerFunc) {
+func (srv *Server) handleHTTPRequest(mux *http.ServeMux, url string, handler http.HandlerFunc) {
 	mux.HandleFunc(url, func(w http.ResponseWriter, req *http.Request) {
-		if !strings.Contains(req.UserAgent(), "Sia-Agent") {
+		if !strings.Contains(req.UserAgent(), "Sia-Agent") && srv.wallet == nil && srv.renter == nil && srv.host == nil && srv.miner == nil {
```

Comment:
oh, bad logic.

desired behavior is to let the request through if the wallet is nil